e2e/benchmarks/services_test.go

```diff
@@ -0,0 +1,276 @@
+// +build e2e
+
+// Runs benchmarks using the Null provider and working with Consul setup with
+// multiple Services or Service Instances.
+
+package benchmarks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/command"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const configFile = "config.hcl"
+const tempDirPrefix = "tmp_"
+
+// Benchmark N Services Instance (all the same service)
+func Benchmark1Instance(b *testing.B) {
+	benchmarkInstances(b, 1)
+}
+func Benchmark100Instances(b *testing.B) {
+	benchmarkInstances(b, 100)
+}
+func Benchmark1000Instances(b *testing.B) {
+	benchmarkInstances(b, 1000)
+}
+func benchmarkInstances(b *testing.B, N int) {
+	srv := testutils.NewTestConsulServerHTTPS(b, "../../testutils")
+	defer srv.Stop()
+	path, cleanup := setupServiceInstances(b, srv, N)
+	defer cleanup() // comment out if you want to keep the config files
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, runSyncOnce(path))
+	}
+}
+
+// Benchmark N Services
+func Benchmark1Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func Benchmark100Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func Benchmark1000Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func benchmarkServices(b *testing.B, N int) {
+	srv := testutils.NewTestConsulServerHTTPS(b, "../../testutils")
+	defer srv.Stop()
+	path, cleanup := setupMultiServices(b, srv, N)
+	defer cleanup() // comment out if you want to keep the config files
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, runSyncOnce(path))
+	}
+}
+
+// ----------------------------------------------------------------------------
+
+// Run CTS in once mode from the cli level
+func runSyncOnce(configPath string) error {
+	cli := command.NewCLI(os.Stdout, os.Stderr)
+	args := []string{"cts", "-once", fmt.Sprintf("--config-file=%s", configPath)}
```

Comment:
Suggestion only if you like to make this 'consul-terraform-sync' instead of 'cts' in case people haven't set up the cts alias on their local development.

Reply:
It isn't actually running the command. It is calling cli.Run(), directly running the code in from the benchmark so it can be profiled. I used 'cts' instead of 'consul-terraform-sync' to help, mistakenly perhaps, call out this difference. Maybe there is something better I can put in there?

Reply:
Ah 🤦 I see! I'm ok with leaving it as is. You also have a clarifying comment on the function already. I think it's just me. I've also been working with running commands lately, so it could be because my head is in that space already. Thanks for clarifying for me!